### PR TITLE
Refactors crematoriums to not use in-world loops.

### PR DESCRIPTION
**Pull request by MarinaGryphon**
Why. *Why.*
Likely needs testing, am on vacation. Should work in theory.

### CODE REVIEW

**Comment by MarinaGryphon** [comment]
!bugfix

**Comment by MarinaGryphon** [comment]
!review

**Comment by Arrow768** [comment]
Doesnt compile.

File: code/game/objects/structures/morgue.dm [hunk 1]
```diff
@@ -422,7 +418,7 @@
 	if(..())
 		return
 	if(src.allowed(user))
-		for (var/obj/structure/crematorium/C in world)
+		for (var/thing in get_listeners_by_type("crematoriums", /obj/structure/crematorium))
```
Comment (MarinaGryphon) [inline code/game/objects/structures/morgue.dm]:
```suggestion
		for (var/obj/structure/crematorium/C in get_listeners_by_type("crematoriums", /obj/structure/crematorium))
```

File: code/game/objects/structures/morgue.dm [hunk 2]
```diff
@@ -188,21 +188,17 @@
...
 
 /obj/structure/crematorium/Destroy()
 	if(connected)
-		qdel(connected)
-		connected = null
-	if(wifi_receiver)
-		qdel(wifi_receiver)
-		wifi_receiver = null
+		QDEL_NULL(connected)
+	if(listener)
```
Comment (MarinaGryphon) [inline code/game/objects/structures/morgue.dm]:
```suggestion
	if(crema)
```

File: code/game/objects/structures/morgue.dm [hunk 3]
```diff
@@ -188,21 +188,17 @@
...
+		QDEL_NULL(listener)
```
Comment (MarinaGryphon) [inline code/game/objects/structures/morgue.dm]:
```suggestion
		QDEL_NULL(crema)
```

**Comment by fernerr** [comment]
Igniters doesn't seem to work after testing them.

**Comment by skull132** [comment]
Re-open when fixed.

**Pull request closed without merging**